apps/console/src/features/server-configurations/pages/connector-listing-page.tsx

```diff
@@ -0,0 +1,433 @@
+            loadCategoryConnectors(connectorCategory.id);
+        });
+    }, [ connectorCategories ]);
+
+    const loadCategoryConnectors = (categoryId: string) => {
+        setConnectorCategoryLoading(true);
+
+        getConnectorCategory(categoryId)
+            .then((response: GovernanceConnectorCategoryInterface) => {
+
+                response?.connectors.map((connector: GovernanceConnectorWithRef) => {
+                    connector.categoryId = categoryId;
+                    connector.ref = React.createRef();
+
+                });
+
+                setConnectors((connectors: GovernanceConnectorWithRef[]) => [
+                    ...connectors, ...response?.connectors as GovernanceConnectorWithRef[]
+                ]);
+                !selectedConnector && setSelectorConnector(response.connectors[ 0 ] as GovernanceConnectorWithRef);
+            })
+            .catch((error: IdentityAppsApiException) => {
+                if (error?.response?.data?.detail) {
+                    dispatch(
+                        addAlert({
+                            description: t(
+                                "console:manage.features.governanceConnectors.notifications." +
+                                "getConnector.error.description",
+                                { description: error.response.data.description }
+                            ),
+                            level: AlertLevels.ERROR,
+                            message: t(
+                                "console:manage.features.governanceConnectors.notifications." +
+                                "getConnector.error.message"
+                            )
+                        })
+                    );
+                } else {
+                    // Generic error message
+                    dispatch(
+                        addAlert({
+                            description: t(
+                                "console:manage.features.governanceConnectors.notifications." +
+                                "getConnector.genericError.description"
+                            ),
+                            level: AlertLevels.ERROR,
+                            message: t(
+                                "console:manage.features.governanceConnectors.notifications." +
+                                "getConnector.genericError.message"
+                            )
+                        })
+                    );
+                }
+            })
+            .finally(() => {
+                setConnectorCategoryLoading(false);
+            });
+    };
+
+    /**
+     * This function returns loading placeholder.
+     */
+    const renderLoadingPlaceholder = (): ReactElement => {
+        return (
+            <Grid.Row columns={ 1 }>
+                <div>
+                    <div 
+                        className="ui card fluid settings-card"
+                        data-testid={ `${testId}-loading-card` }
```

Comment:
let's add data-componentid since data-testid is deprecated.

Reply:
this is an existing test id not introduced by this PR